### ISSUE TITLE
Crash on Mixpanel track:properties:

### ISSUE THREAD

**Issue by razelkayamdev** [mixpanel/mixpanel-iphone#707]
**Integration Method:** CocoaPods
**Xcode Version:** Version 8.3.3 (8E3004b)
**Library Version:** 3.1.8
**Platform:** iOS 
**Language:** Swift
**Description:**  Crash, occurred once, to one user. Not reproducible.
**Expected Behavior:** Not crash.

 Organization: TheCultureTrip
 Platform: ios
 Application: Culture Trip
 Version: N/A
 Bundle Identifier: com.theculturetrip
 Issue #: 392
 Issue ID: 596a6d05be077a4dcc9ccf9d
 Session ID: abe06d761b3d4cef8ddeb440490633a9_e0b90408699311e7acae56847afe9799_0_v2
 Date: 2017-07-15T19:28:00Z
 OS Version: 9.2.0 (13C75)
 Device: iPhone 5s
 RAM Free: 4.3%
 Disk Free: 7.8%

#4. Crashed: com.mixpanel.218d236bed7834fd72edf1f852fd7e6a.0x13103f3e0
0  libobjc.A.dylib                0x1811b9bdc objc_msgSend + 28
1  Mixpanel                       0x10085d1e8 __29-[Mixpanel track:properties:]_block_invoke (Mixpanel.m:412)
2  libdispatch.dylib              0x18158d630 _dispatch_call_block_and_release + 24
3  libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
4  libdispatch.dylib              0x181599634 _dispatch_queue_drain + 864
5  libdispatch.dylib              0x1815910f4 _dispatch_queue_invoke + 464
6  libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
7  libdispatch.dylib              0x18159ba88 _dispatch_root_queue_drain + 2140
8  libdispatch.dylib              0x18159b224 _dispatch_worker_thread3 + 112
9  libsystem_pthread.dylib        0x1817a1470 _pthread_wqthread + 1092
10 libsystem_pthread.dylib        0x1817a1020 start_wqthread + 4

--

#0. com.apple.main-thread
0  libsystem_kernel.dylib         0x1816c438c read + 8
1  CoreFoundation                 0x181a50e10 _CFReadBytesFromFile + 296
2  CoreFoundation                 0x181a4fb74 CFURLCreateDataAndPropertiesFromResource + 548
3  CoreFoundation                 0x181b1efac _CFDataCreateFromURL + 72
4  CoreFoundation                 0x181b8be8c CFBundleCopyLocalizedStringForLocalization + 328
5  Foundation                     0x182419a60 -[NSBundle localizedStringForKey:value:table:] + 60
6  libswiftFoundation.dylib       0x100e5c758 _TTSf4gs_g_g_gs_d___TF10Foundation17NSLocalizedStringFTSS9tableNameGSqSS_6bundleCSo6Bundle5valueSS7commentSS_SS + 180
7  libswiftFoundation.dylib       0x100dfc1f4 _TF10Foundation17NSLocalizedStringFTSS9tableNameGSqSS_6bundleCSo6Bundle5valueSS7commentSS_SS + 80
8  CultureTrip                    0x1000d9d60 OnboardingViewController.setupTexts() -> () (OnboardingViewController.swift)
9  CultureTrip                    0x1000d96f0 OnboardingViewController.viewDidLoad() -> () (OnboardingViewController.swift)
10 CultureTrip                    0x1000d9774 @objc OnboardingViewController.viewDidLoad() -> () (OnboardingViewController.swift)
11 UIKit                          0x18682c0c0 -[UIViewController loadViewIfRequired] + 996
12 UIKit                          0x1868ebda8 -[UINavigationController _layoutViewController:] + 72
13 UIKit                          0x1868ebc80 -[UINavigationController _updateScrollViewFromViewController:toViewController:] + 416
14 UIKit                          0x1868eaec8 -[UINavigationController _startTransition:fromViewController:toViewController:] + 144
15 UIKit                          0x1868eaa6c -[UINavigationController _startDeferredTransitionIfNeeded:] + 868
16 UIKit                          0x1868ea694 -[UINavigationController __viewWillLayoutSubviews] + 60
17 UIKit                          0x1868ea5fc -[UILayoutContainerView layoutSubviews] + 208
18 UIKit                          0x186827778 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 656
19 QuartzCore                     0x184236b2c -[CALayer layoutSublayers] + 148
20 QuartzCore                     0x184231738 CA::Layer::layout_if_needed(CA::Transaction*) + 292
21 QuartzCore                     0x1842315f8 CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 32
22 QuartzCore                     0x184230c94 CA::Context::commit_transaction(CA::Transaction*) + 252
23 QuartzCore                     0x1842309dc CA::Transaction::commit() + 512
24 QuartzCore                     0x18422a0cc CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*) + 80
25 CoreFoundation                 0x181af0588 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 32
26 CoreFoundation                 0x181aee32c __CFRunLoopDoObservers + 372
27 CoreFoundation                 0x181a1d6a0 CFRunLoopRunSpecific + 416
28 UIKit                          0x18689a580 -[UIApplication _run] + 460
29 UIKit                          0x186894d90 UIApplicationMain + 204
30 CultureTrip                    0x1000ca1ac main (RecentSearch.swift:15)
31 libdispatch.dylib              0x1815be8b8 (Missing)

#1. Thread
0  libsystem_pthread.dylib        0x1817a101c start_wqthread + 14

#2. com.apple.libdispatch-manager
0  libsystem_kernel.dylib         0x1816dd4fc kevent_qos + 8
1  libdispatch.dylib              0x1815a094c _dispatch_mgr_invoke + 232
2  libdispatch.dylib              0x18158f7bc _dispatch_source_invoke + 50

#3. Thread
0  libsystem_pthread.dylib        0x1817a101c start_wqthread + 14

#4. Crashed: com.mixpanel.218d236bed7834fd72edf1f852fd7e6a.0x13103f3e0
0  libobjc.A.dylib                0x1811b9bdc objc_msgSend + 28
1  Mixpanel                       0x10085d1e8 __29-[Mixpanel track:properties:]_block_invoke (Mixpanel.m:412)
2  libdispatch.dylib              0x18158d630 _dispatch_call_block_and_release + 24
3  libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
4  libdispatch.dylib              0x181599634 _dispatch_queue_drain + 864
5  libdispatch.dylib              0x1815910f4 _dispatch_queue_invoke + 464
6  libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
7  libdispatch.dylib              0x18159ba88 _dispatch_root_queue_drain + 2140
8  libdispatch.dylib              0x18159b224 _dispatch_worker_thread3 + 112
9  libsystem_pthread.dylib        0x1817a1470 _pthread_wqthread + 1092
10 libsystem_pthread.dylib        0x1817a1020 start_wqthread + 4

#5. Thread
0  libsystem_kernel.dylib         0x1816dcb6c __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1817a1530 _pthread_wqthread + 1284
2  libsystem_pthread.dylib        0x1817a1020 start_wqthread + 4

#6. com.twitter.crashlytics.ios.MachExceptionServer
0  CultureTrip                    0x1002060c0 CLSProcessRecordAllThreads (CLSProcess.c:376)
1  CultureTrip                    0x1002060c0 CLSProcessRecordAllThreads (CLSProcess.c:376)
2  CultureTrip                    0x100205f70 CLSProcessRecordAllThreads (CLSProcess.c:407)
3  CultureTrip                    0x1001f6214 CLSHandler (CLSHandler.m:26)
4  CultureTrip                    0x1001f11dc CLSMachExceptionServer (CLSMachException.c:446)
5  libsystem_pthread.dylib        0x1817a3b28 _pthread_body + 156
6  libsystem_pthread.dylib        0x1817a3a8c _pthread_body + 154
7  libsystem_pthread.dylib        0x1817a1028 thread_start + 4

#7. NSOperationQueue 0x131104ba0 :: NSOperation 0x13111e9c0 (QOS: LEGACY)
0  libsystem_kernel.dylib         0x1816c14f8 semaphore_wait_trap + 8
1  libdispatch.dylib              0x18159e55c _dispatch_semaphore_wait_slow + 244
2  CFNetwork                      0x18216a33c CFURLConnectionSendSynchronousRequest + 288
3  CFNetwork                      0x18218d92c +[NSURLConnection sendSynchronousRequest:returningResponse:error:] + 120
4  CultureTrip                    0x1000be21c __70-[SPEmitter sendEventWithRequest:andIndex:andResultArray:andOversize:]_block_invoke (SPEmitter.m:320)
5  Foundation                     0x1824e0334 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
6  Foundation                     0x182433100 -[NSBlockOperation main] + 96
7  Foundation                     0x182423348 -[__NSOperationInternal _start:] + 604
8  Foundation                     0x1824e2728 __NSOQSchedule_f + 224
9  libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
10 libdispatch.dylib              0x181599634 _dispatch_queue_drain + 864
11 libdispatch.dylib              0x1815910f4 _dispatch_queue_invoke + 464
12 libdispatch.dylib              0x18159b504 _dispatch_root_queue_drain + 728
13 libdispatch.dylib              0x18159b224 _dispatch_worker_thread3 + 112
14 libsystem_pthread.dylib        0x1817a1470 _pthread_wqthread + 1092
15 libsystem_pthread.dylib        0x1817a1020 start_wqthread + 4

#8. Thread
0  libsystem_kernel.dylib         0x1816dcb6c __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x1817a1530 _pthread_wqthread + 1284
2  libsystem_pthread.dylib        0x1817a1020 start_wqthread + 4

#9. com.apple.root.default-qos
0  libsystem_kernel.dylib         0x1816dbf48 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x1817a2ce8 _pthread_cond_wait + 648
2  Foundation                     0x1824de6e0 -[__NSOperationInternal _waitUntilFinished:] + 132
3  Foundation                     0x182480128 -[NSOperationQueue waitUntilAllOperationsAreFinished] + 252
4  CultureTrip                    0x1000bdea8 -[SPEmitter sendEvents] (SPEmitter.m:279)
5  libdispatch.dylib              0x18158d630 _dispatch_call_block_and_release + 24
6  libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
7  libdispatch.dylib              0x18159ba88 _dispatch_root_queue_drain + 2140
8  libdispatch.dylib              0x18159b224 _dispatch_worker_thread3 + 112
9  libsystem_pthread.dylib        0x1817a1470 _pthread_wqthread + 1092
10 libsystem_pthread.dylib        0x1817a1020 start_wqthread + 4

#10. com.apple.CoreLocation.ConnectionClient.0x12ffd6370.events
0  libsystem_kernel.dylib         0x1816c1510 semaphore_timedwait_trap + 8
1  libdispatch.dylib              0x18159e508 _dispatch_semaphore_wait_slow + 160
2  CoreLocation                   0x1883aad78 (null) + 628
3  CoreLocation                   0x1883af7ec (null) + 1176
4  CoreLocation                   0x1883e9f28 (null) + 59072
5  CoreLocation                   0x1883e7b7c (null) + 49940
6  CoreLocation                   0x1883e7ab0 (null) + 49736
7  CoreLocation                   0x1883e8cfc (null) + 54420
8  libxpc.dylib                   0x1817d5358 _xpc_connection_call_event_handler + 68
9  libxpc.dylib                   0x1817d3100 _xpc_connection_mach_event + 2132
10 libdispatch.dylib              0x18158d6ec _dispatch_client_callout4 + 16
11 libdispatch.dylib              0x1815919d8 _dispatch_mach_msg_invoke + 656
12 libdispatch.dylib              0x181599808 _dispatch_queue_drain + 1332
13 libdispatch.dylib              0x181590aec _dispatch_mach_invoke + 600
14 libdispatch.dylib              0x181599808 _dispatch_queue_drain + 1332
15 libdispatch.dylib              0x1815910f4 _dispatch_queue_invoke + 464
16 libdispatch.dylib              0x181599808 _dispatch_queue_drain + 1332
17 libdispatch.dylib              0x1815910f4 _dispatch_queue_invoke + 464
18 libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
19 libdispatch.dylib              0x18159ba88 _dispatch_root_queue_drain + 2140
20 libdispatch.dylib              0x18159b224 _dispatch_worker_thread3 + 112
21 libsystem_pthread.dylib        0x1817a1470 _pthread_wqthread + 1092
22 libsystem_pthread.dylib        0x1817a1020 start_wqthread + 4

#11. com.mixpanel.218d236bed7834fd72edf1f852fd7e6a.0x13103f3e0.network
0  libsystem_kernel.dylib         0x1816c14f8 semaphore_wait_trap + 8
1  libdispatch.dylib              0x18159e55c _dispatch_semaphore_wait_slow + 244
2  Mixpanel                       0x100862fb4 __58-[Mixpanel checkForDecideResponseWithCompletion:useCache:]_block_invoke (Mixpanel.m:1512)
3  libdispatch.dylib              0x18158d630 _dispatch_call_block_and_release + 24
4  libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
5  libdispatch.dylib              0x181599634 _dispatch_queue_drain + 864
6  libdispatch.dylib              0x1815910f4 _dispatch_queue_invoke + 464
7  libdispatch.dylib              0x18158d5f0 _dispatch_client_callout + 16
8  libdispatch.dylib              0x18159ba88 _dispatch_root_queue_drain + 2140
9  libdispatch.dylib              0x18159b224 _dispatch_worker_thread3 + 112
10 libsystem_pthread.dylib        0x1817a1470 _pthread_wqthread + 1092
11 libsystem_pthread.dylib        0x1817a1020 start_wqthread + 4

#12. com.apple.NSURLConnectionLoader
0  libcorecrypto.dylib            0x18156c1c4 cczp_mul_ws + 44
1  libcorecrypto.dylib            0x181546a34 XYCZadd + 340
2  libcorecrypto.dylib            0x181546a34 XYCZadd + 340
3  libcorecrypto.dylib            0x1815464a8 ccec_mult + 1012
4  libcorecrypto.dylib            0x18156f330 ccec_generate_key_internal_fips + 316
5  libcorecrypto.dylib            0x181547758 ccec_generate_key_fips + 24
6  libsystem_coretls.dylib        0x18167c974 sslEcdhCreateKey + 76
7  libsystem_coretls.dylib        0x181680700 SSLEncodeKeyExchange + 1112
8  libsystem_coretls.dylib        0x181683170 SSLPrepareAndQueueMessage + 48
9  libsystem_coretls.dylib        0x181681f5c SSLAdvanceHandshake + 1792
10 libsystem_coretls.dylib        0x181686754 tls_handshake_continue + 36
11 Security                       0x181f479ac SSLHandshakeProceed + 80
12 Security                       0x181f47e10 SSLHandshake + 140
13 CFNetwork                      0x182111784 SocketStream::_PerformSecurityHandshake_NoLock() + 464
14 CFNetwork                      0x1820ec244 SocketStream::socketCallback(__CFSocket*, unsigned long, __CFData const*, void const*) + 240
15 CFNetwork                      0x1820ec110 SocketStream::_SocketCallBack_stream(__CFSocket*, unsigned long, __CFData const*, void const*, void*) + 88
16 CoreFoundation                 0x181af5628 __CFSocketPerformV0 + 1216
17 CoreFoundation                 0x181af0efc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
18 CoreFoundation                 0x181af0990 __CFRunLoopDoSources0 + 540
19 CoreFoundation                 0x181aee690 __CFRunLoopRun + 724
20 CoreFoundation                 0x181a1d680 CFRunLoopRunSpecific + 384
21 CFNetwork                      0x18218d434 +[NSURLConnection(Loader) _resourceLoadLoop:] + 412
22 Foundation                     0x1824fbc40 __NSThread__start__ + 1000
23 libsystem_pthread.dylib        0x1817a3b28 _pthread_body + 156
24 libsystem_pthread.dylib        0x1817a3a8c _pthread_body + 154
25 libsystem_pthread.dylib        0x1817a1028 thread_start + 4

#13. com.apple.CFSocket.private
0  libsystem_kernel.dylib         0x1816dc368 __select + 8
1  CoreFoundation                 0x181af7028 __CFSocketManager + 648
2  libsystem_pthread.dylib        0x1817a3b28 _pthread_body + 156
3  libsystem_pthread.dylib        0x1817a3a8c _pthread_body + 154
4  libsystem_pthread.dylib        0x1817a1028 thread_start + 4

**Comment by sairasamdanii** [comment]
I'm facing same error. My app is crashing on setting mixpanel properties. And crash is in following function

- (NSString *)mp_controllerVariable    in   UIView+MPHelpers.m

Following is the statement where execution stops

if (ivar_getTypeEncoding(ivar)[0] == '@' && object_getIvar(responder, ivar) == self)

and gives EXC_BAD_ACCESS with following log:

* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=EXC_I386_GPFLT)
    frame #0: 0x00000001163cacea libobjc.A.dylib`objc_retain + 10
  * frame #1: 0x000000010f47a4ef FalconExpense`-[UIView(self=0x00007fbb61d47ca0, _cmd="mp_controllerVariable") mp_controllerVariable] at UIView+MPHelpers.m:90
    frame #2: 0x000000010f47b0c4 FalconExpense`-[UIView(self=0x00007fbb61d47ca0, _cmd="mp_varB") mp_varB] at UIView+MPHelpers.m:185
    frame #3: 0x00000001117d0f9d Foundation`-[NSObject(NSKeyValueCoding) valueForKey:] + 284
    frame #4: 0x00000001117cf0ad Foundation`-[NSFunctionExpression expressionValueWithObject:context:] + 731
    frame #5: 0x00000001117d02f8 Foundation`-[NSComparisonPredicate evaluateWithObject:substitutionVariables:] + 239
    frame #6: 0x00000001117d0129 Foundation`-[NSCompoundPredicateOperator evaluatePredicates:withObject:substitutionVariables:] + 541
    frame #7: 0x00000001117cfd88 Foundation`-[NSCompoundPredicate evaluateWithObject:substitutionVariables:] + 289
    frame #8: 0x000000010f449dd2 FalconExpense`__30-[MPObjectSelector nextFilter]_block_invoke((null)=0x0000600000251cd0, evaluatedObject=0x00007fbb61d47ca0, bindings=0x0000000000000000) at MPObjectSelector.m:174
    frame #9: 0x0000000111835a94 Foundation`-[NSBlockPredicate evaluateWithObject:substitutionVariables:] + 211
    frame #10: 0x000000010f44b1e5 FalconExpense`-[MPObjectFilter appliesTo:](self=0x0000600000251f70, _cmd="appliesTo:", view=0x00007fbb61d47ca0) at MPObjectSelector.m:303
    frame #11: 0x000000010f44b4f1 FalconExpense`-[MPObjectFilter appliesToAny:](self=0x0000600000251f70, _cmd="appliesToAny:", views=@"1 element") at MPObjectSelector.m:315
    frame #12: 0x000000010f44925f FalconExpense`-[MPObjectSelector isLeafSelected:fromRoot:evaluatingFinalPredicate:](self=0x0000600000281d60, _cmd="isLeafSelected:fromRoot:evaluatingFinalPredicate:", leaf=0x00007fbb61d47ca0, root=0x00007fbb63889a00, finalPredicate=YES) at MPObjectSelector.m:131
    frame #13: 0x000000010f449011 FalconExpense`-[MPObjectSelector isLeafSelected:fromRoot:](self=0x0000600000281d60, _cmd="isLeafSelected:fromRoot:", leaf=0x00007fbb61d47ca0, root=0x00007fbb63889a00) at MPObjectSelector.m:115
    frame #14: 0x000000010f46685e FalconExpense`+[MPVariantAction executeSelector:withArgs:onPath:fromRoot:toLeaf:](self=MPVariantAction, _cmd="executeSelector:withArgs:onPath:fromRoot:toLeaf:", selector="setUserInteractionEnabled:", args=@"1 element", path=0x0000600000281d60, root=0x00007fbb63889a00, leaf=0x00007fbb61d47ca0) at MPVariant.m:545
    frame #15: 0x000000010f4651dd FalconExpense`__26-[MPVariantAction execute]_block_invoke_2((null)=<unavailable>) at MPVariant.m:442
    frame #16: 0x000000010f476ea1 FalconExpense`+[NSThread(self=NSThread, _cmd="mp_safelyRunOnMainThreadSync:", block=0x000000010f464ff0) mp_safelyRunOnMainThreadSync:] at NSThread+MPHelpers.m:15
    frame #17: 0x000000010f464fae FalconExpense`__26-[MPVariantAction execute]_block_invoke((null)=0x000060800005c680, view=0x00007fbb61d47ca0, command="didMoveToWindow") at MPVariant.m:437
    frame #18: 0x000000010f456c0e FalconExpense`mp_swizzledMethod_2(self=0x00007fbb61d47ca0, _cmd="didMoveToWindow") at MPSwizzler.m:45
    frame #19: 0x0000000112b8df01 UIKit`-[UIView(Internal) _didMoveFromWindow:toWindow:] + 1737
    frame #20: 0x0000000112c56a3d UIKit`-[UIControl _didMoveFromWindow:toWindow:] + 67
    frame #21: 0x0000000112b8db43 UIKit`-[UIView(Internal) _didMoveFromWindow:toWindow:] + 779
    frame #22: 0x0000000112b8db43 UIKit`-[UIView(Internal) _didMoveFromWindow:toWindow:] + 779
    frame #23: 0x0000000112ba4fc1 UIKit`-[UIScrollView _didMoveFromWindow:toWindow:] + 84
    frame #24: 0x0000000112b8db43 UIKit`-[UIView(Internal) _didMoveFromWindow:toWindow:] + 779
    frame #25: 0x0000000112b80b96 UIKit`__45-[UIView(Hierarchy) _postMovedFromSuperview:]_block_invoke + 151
    frame #26: 0x0000000112b80a7d UIKit`-[UIView(Hierarchy) _postMovedFromSuperview:] + 828
    frame #27: 0x0000000112b7e77d UIKit`__UIViewWasRemovedFromSuperview + 172
    frame #28: 0x0000000112b7e27f UIKit`-[UIView(Hierarchy) removeFromSuperview] + 576
    frame #29: 0x0000000112caf6cb UIKit`-[UINavigationController navigationTransitionView:didEndTransition:fromView:toView:] + 591
    frame #30: 0x0000000112ca52d4 UIKit`__49-[UINavigationController _startCustomTransition:]_block_invoke + 246
    frame #31: 0x0000000113600d1c UIKit`-[_UIViewControllerTransitionContext completeTransition:] + 102
    frame #32: 0x0000000112ab93c0 UIKit`__53-[_UINavigationParallaxTransition animateTransition:]_block_invoke.99 + 803
    frame #33: 0x0000000112b8a024 UIKit`-[UIViewAnimationBlockDelegate _didEndBlockAnimation:finished:context:] + 527
    frame #34: 0x0000000112b5d257 UIKit`-[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 222
    frame #35: 0x0000000112b5d796 UIKit`-[UIViewAnimationState animationDidStop:finished:] + 136
    frame #36: 0x000000011269068e QuartzCore`CA::Layer::run_animation_callbacks(void*) + 306
    frame #37: 0x000000011708405c libdispatch.dylib`_dispatch_client_callout + 8
    frame #38: 0x000000011706540b libdispatch.dylib`_dispatch_main_queue_callback_4CF + 411
    frame #39: 0x0000000116917909 CoreFoundation`__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
    frame #40: 0x00000001168ddae4 CoreFoundation`__CFRunLoopRun + 2164
    frame #41: 0x00000001168dd016 CoreFoundation`CFRunLoopRunSpecific + 406
    frame #42: 0x0000000117f2da24 GraphicsServices`GSEventRunModal + 62
    frame #43: 0x0000000112ad1134 UIKit`UIApplicationMain + 159
    frame #44: 0x000000010ef47f8f FalconExpense`main(argc=1, argv=0x00007fff50cb8588) at main.m:16
    frame #45: 0x00000001170d065d libdyld.dylib`start + 1
    frame #46: 0x00000001170d065d libdyld.dylib`start + 1

**Comment by pchien** [comment]
@sairasamdanii this isn't when you're tracking right? Does this just happen randomly or is it when you're connected to AB testing?

**Comment by sairasamdanii** [comment]
It happens when I call track or se properties on mixpanel people

**Comment by pchien** [comment]
@sairasamdanii it looks like your crash is A/B testing related. What iOS version are you running when it crashes? What version of the SDK are you using? And can I your Mixpanel project id (the number after `mixpanel.com/report/` in the url) so I can take a look at your A/B test? Don't worry about posting it publicly, non-Mixpanel employees won't be able to use the project id for anything. Thanks!

**Comment by sairasamdanii** [comment]
@pchien Well, AB Testing is enabled for this project. I've tried running it
on different iOS version, it crashes on all. Mixpanel SDK version is
3.1.9.  Mixpanel project id is 235788

**Comment by sairasamdanii** [comment]
@pchien Any response?

**Comment by pchien** [comment]
Sorry @sairasamdanii I haven't had time to look into it yet. I'll definitely take a look before the end of the week.

**Comment by pchien** [comment]
@sairasamdanii do you know what object is causing the bad access on that line? As far as I can tell, the `responder` has to be okay there because there's already a strong reference from a local variable in the function. The `ivar` itself is fine unless if `class_copyIvarList` can return invalid ivars. So it's either the return value of `object_getIvar` or `self` that's an issue. I suspect it's the return value of `object_getIvar` because `self` is used previously in the function. This code iterates over properties of objects in your app. Is there anywhere where you have `unsafe_unretained` properties or otherwise could have some properties still around that were deallocated? With ARC it's generally recommended that all your properties be `weak` or `strong` to avoid issues like this.

**Comment by sairasamdanii** [comment]
I could not understand which properties you are referencing to? The
properties that I'm setting to mixpanel people?

**Comment by pchien** [comment]
The line you're crashing on is inside a loop where, for A/B testing to identify views, it loops over ivars/properties (Obj-C properties, not Mixpanel ones) of UIResponders. Next time it crashes, can you spend a bit of time debugging to see what exactly is causing the bad access? It's hard for me to say what the issue is without being able to see what's happening when the exception breakpoint is hit. I suspect there's something wrong with the object that gets returned from `object_getIvar` but I'm not 100% sure, and that's an object from your app, not from Mixpanel's SDK.

**Comment by sairasamdanii** [comment]
@pchien I have debugged it. And it stops at a button. As it happens in multiple classes, so in each class, there is any of the buttons on which it crashes. My issue is that button is apparently working fine. But as soon as i call mixpanel track and set people properties and after that I want to move back to previous screen, it crashes on that specific button.

**Comment by sairasamdanii** [comment]
@pchien I'm getting crash on `object_getIvar(responder, ivar) == self` in class `UIView+MPHelpers`
I have seen another issue related to this but that is closed with no proper result. #362 

Please tell me will that be helpful in any way. Is it related to UIButton some how.

**Comment by pchien** [comment]
@sairasamdanii so `self` is a button in this case? What's `responder`? And what's the `ivar`? Both name and the type of object would be useful. I'm curious if the ivar is accessible and it's crashing on `self` or the other way around?

Also, if it only crashes after you track and set properties in Mixpanel. Can you try just initializing Mixpanel but not tracking or setting properties? I thought it would crash if you initialize Mixpanel because A/B tests start running after initialization.

**Comment by sairasamdanii** [comment]
@pchien self is button. responder is my view controller. ivar is not nil. It passes check 
`ivar_getTypeEncoding(ivar)[0] == '@'` 
But crashes on this statement:
`object_getIvar(responder, ivar)`

I've attached screenshot too.
![crash on mixpanel](https://user-images.githubusercontent.com/7940521/29383418-c1809b4a-82e9-11e7-8c9d-d8579c03525d.png)

I've tried commenting out the mixpanel track and set properties code and just initialized mixpanel using following statement and it threw similar crash on that too.

`Mixpanel *mixpanel = [Mixpanel sharedInstance];`

**Comment by sairasamdanii** [comment]
@pchien My project is in very crucial stage and we have to upload new release ASAP but are stuck due to this issue. Your help will be highly appreciated.

**Comment by pchien** [comment]
@sairasamdanii I think that means that the object that it's trying to get from your view controller (`responder`) is deallocated or otherwise pointing to bad memory. You can use `ivar_getName(ivar)` to try to see what it is, and maybe that can tell you what's going on. I think Mixpanel is just exposing a memory issue within your app because Mixpanel is crawling through the view hierarchy and inspects properties on view controllers while doing so.

**Comment by sairasamdanii** [comment]
@pchien my concern is why its the same object. In attached screenshot, you can see that object exists. I can even print title label of that button. And it shows correct title. If that view is deallocated, either all of its subviews should be deallocated or it should be random views not one specific button.

Secondly, If I keep strong reference of that object, still I get same crash.

Thirdly, I have not changed my code. This code was working perfectly fine with mixpanel before. This crash has started coming only after latest update of mixpanel. Has mixpanel made some changes?

**Comment by pchien** [comment]
I think it isn't the button that's causing the crash. It's whatever is inside the ivar (the result of `object_getIvar`). We've made a couple minor changes to A/B testing recently but I don't think we've touched this code at all.

**Comment by yarneo** [comment]
Just a side suggestion @sairasamdanii , if you are in a time crunch and noticed that a previous Mixpanel release worked well for you and didn't crash, then I would point to that release for now and we can try to help you resolve this issue for your future releases.

**Comment by sairasamdanii** [comment]
/@pchien is there any way to debug that what's inside ivar that causes crash. Because each time the crash happens, self is this specific button

**Comment by pchien** [comment]
@sairasamdanii you know what `responder` is and you can get the name of the ivar with `ivar_getName`. That's the ivar on your object `responder`. Then you can inspect/debug what is going on with that object before it crashes. You can't look at what's in it at the time of the crash because trying to look at it is what's causing the crash.

**Comment by pchien** [comment]
Closing this because I assume no activity means the issue was resolved.
